pkg/tsdb/elasticsearch/client/client.go

```
@@ -323,3 +337,173 @@ func (c *baseClientImpl) MultiSearch() *MultiSearchRequestBuilder {
+	}
+
+	if !c.ds.BasicAuth && c.ds.User != "" {
+		clientLog.Debug("Request configured to use basic authentication")
+		req.SetBasicAuth(c.ds.User, c.ds.DecryptedPassword())
+	}
+
+	httpClient, err := newDatasourceHttpClient(c.ds)
+	if err != nil {
+		return nil, err
+	}
+
+	start := time.Now()
+	defer func() {
+		elapsed := time.Since(start)
+		clientLog.Debug("Executed request", "took", elapsed)
+	}()
+	//nolint:bodyclose
+	resp, err := ctxhttp.Do(c.ctx, httpClient, req)
+	if err != nil {
+		return nil, err
+	}
+	return &pplresponse{
+		httpResponse: resp,
+		reqInfo:      reqInfo,
+	}, nil
+}
+
+func (c *baseClientImpl) ExecutePPLQuery(r *PPLRequest) (*PPLResponse, error) {
+	clientLog.Debug("Executing PPL", "PPL requests")
+
+	req := createPPLRequest(r)
+	clientRes, err := c.executePPLRequest("_opendistro/_ppl", req)
+	if err != nil {
+		return nil, err
+	}
+	res := clientRes.httpResponse
+	defer res.Body.Close()
+
+	clientLog.Debug("Received PPL response", "code", res.StatusCode, "status", res.Status, "content-length", res.ContentLength)
+
+	start := time.Now()
+	clientLog.Debug("Decoding PPL json response")
+
+	var bodyBytes []byte
+	if c.debugEnabled {
+		tmpBytes, err := ioutil.ReadAll(res.Body)
+		if err != nil {
+			clientLog.Error("failed to read http response bytes", "error", err)
+		} else {
+			bodyBytes = make([]byte, len(tmpBytes))
+			copy(bodyBytes, tmpBytes)
+			res.Body = ioutil.NopCloser(bytes.NewBuffer(tmpBytes))
+		}
+	}
+
+	var pr PPLResponse
+	dec := json.NewDecoder(res.Body)
+	err = dec.Decode(&pr)
+	if err != nil {
+		return nil, err
+	}
+
+	elapsed := time.Since(start)
+	clientLog.Debug("Decoded PPL json response", "took", elapsed)
+
+	pr.Status = res.StatusCode
+
+	if c.debugEnabled {
+		bodyJSON, err := simplejson.NewFromReader(bytes.NewBuffer(bodyBytes))
+		var data *simplejson.Json
+		if err != nil {
+			clientLog.Error("failed to decode http response into json", "error", err)
```

Shouldn't we return an error if this case happens at line 497?